python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -768,63 +771,66 @@ def contains_properly(self, other, align=True):
-            geom = rhs.polygons
-        elif contains_only_multipoints(rhs) is True:
-            # mpoint in polygon
-            mode = "MULTIPOINTS"
-            geom = rhs.multipoints
-        else:
-            # no conditioning is required
-            geom = rhs.points
-        xy_points = geom.xy
-        point_indices = geom.point_indices()
-        points = GeoSeries(GeoColumn._from_points_xy(xy_points._column)).points
-
-        # call pip on the three subtypes on the right:
-        point_result = contains_properly(
-            points.x,
-            points.y,
-            lhs.polygons.part_offset[:-1],
-            lhs.polygons.ring_offset[:-1],
-            lhs.polygons.x,
-            lhs.polygons.y,
-        )
-        if (
-            mode == "LINESTRINGS"
-            or mode == "POLYGONS"
-            or mode == "MULTIPOINTS"
-        ):
-            # process for completed linestrings, polygons, and multipoints.
-            # Not necessary for points.
-            result = cudf.DataFrame(
-                {"idx": point_indices, "pip": point_result}
-            )
-            # if the number of points in the polygon is equal to the number of
-            # points, then the requirements for `.contains_properly` are met
-            # for this geometry type.
-            df_result = (
-                result.groupby("idx").sum().sort_index()
-                == result.groupby("idx").count().sort_index()
-            )
-            point_result = cudf.Series(
-                df_result["pip"], index=cudf.RangeIndex(0, len(df_result))
-            )
-            point_result.name = None
-        return point_result
+    def intersects(self, other, align=True):
+        """Compute the intersections of two GeoSeries.
+
+        Parameters
+        ----------
+        other
+            a cuspatial.GeoSeries
+        align=True
+            align the GeoSeries indexes before calling the binpred
+
+        Returns
+        -------
+        result : cudf.Series
+            A Series of boolean values indicating whether the geometries of
+            each row intersect.
+        """
+        return IntersectsBinpred(self, other, align)()
+
+    def within(self, other, align=True):
+        """An object is said to be within other if at least one of its points
+        is located in the interior and no points are located in the exterior
+        of the other.
+
+        Parameters
+        ----------
+        other
+            a cuspatial.GeoSeries
+        align=True
+            align the GeoSeries indexes before calling the binpred
+
+        Returns
+        -------
+        result : cudf.Series
+            A Series of boolean values indicating whether each feature falls
+            within the corresponding polygon in the input.
+        """
+        return WithinBinpred(self, other, align)()
+
+    def overlaps(self, other, align=True):
+        """Returns True for all aligned geometries that overlap other, else
```

Comment:
This one starts with saying what it returns, then defines overlap.

For all of my comments that start with "this one", please make the documentation consistent. Please just emulate the GeoPandas docs. :)